plugins/terminal.koplugin/main.lua

```diff
@@ -1,58 +1,394 @@
+local ButtonDialog = require("ui/widget/buttondialog")
+local CenterContainer = require("ui/widget/container/centercontainer")
 local DataStorage = require("datastorage")
 local Font = require("ui/font")
 local InfoMessage = require("ui/widget/infomessage")
 local InputDialog = require("ui/widget/inputdialog")
+local LuaSettings = require("luasettings")
+local Menu = require("ui/widget/menu")
+local Screen = require("device").screen
 local TextViewer = require("ui/widget/textviewer")
 local Trapper = require("ui/trapper")
 local UIManager = require("ui/uimanager")
 local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local logger = require("logger")
 local util = require("ffi/util")
 local _ = require("gettext")
-local Screen = require("device").screen
 
 local Terminal = WidgetContainer:new{
     name = "terminal",
-    dump_file = util.realpath(DataStorage:getDataDir()) .. "/terminal_output.txt",
     command = "",
+    dump_file = util.realpath(DataStorage:getDataDir()) .. "/terminal_output.txt",
+    items_per_page = G_reader_settings:readSetting("items_per_page") or 16,
+    settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/terminal_shortcuts.lua"),
+    shortcuts_dialog = nil,
+    shortcuts_menu = nil,
+    --    shortcuts_file = DataStorage:getSettingsDir() .. "/terminal_shortcuts.lua",
+    shortcuts = {},
+    source = "terminal",
 }
 
 function Terminal:init()
     self.ui.menu:registerToMainMenu(self)
+    self.shortcuts = self.settings:readSetting("shortcuts") or {}
+end
+
+function Terminal:saveShortcuts()
+    self.settings:saveSetting("shortcuts", self.shortcuts)
+    self.settings:flush()
+    UIManager:show(InfoMessage:new {
+        text = _("Shortcuts saved!"),
```

Comment:
As you must have noticed, we're some rather boring and not very enthusiastic people - and we made a point to not have any enthusiastic message. So, no `!` :) Just a plain `Shorcuts saved.` (Really, we don't have any, except one in the goodreads plugin: "Search not found!")